tests/core/mempool/test_mempool_manager.py

```diff
@@ -213,7 +247,121 @@ async def get_coin_record(coin_id: bytes32) -> Optional[CoinRecord]:
+    sb = SpendBundle.aggregate([sb1, sb2])
+    mempool_item = mempool_item_from_spendbundle(sb)
+    saved_cost = uint64(44)
+    created_coins = {Coin(TEST_COIN_ID, IDENTITY_PUZZLE_HASH, 1), Coin(TEST_COIN_ID, IDENTITY_PUZZLE_HASH, 2)}
+    expected_new_dedups = {
+        TEST_COIN_ID: DedupCoinSpend(solution=initial_solution, cost=saved_cost, additions=created_coins),
+        TEST_COIN_ID2: DedupCoinSpend(solution=second_solution, cost=None, additions=set()),
+    }
+    spends_to_dedup, cost_saving, new_dedups, dedup_additions = find_duplicate_spends(mempool_item, dedup_coin_spends)
+    # Only the eligible one that we encountered more than once gets deduplicated
+    assert spends_to_dedup == sb1.coin_spends
+    assert cost_saving == saved_cost
+    assert new_dedups == expected_new_dedups
+    assert dedup_additions == created_coins
+
+
+def test_find_duplicate_spends_eligible_3rd_time_another_2nd_time_and_one_non_eligible() -> None:
+    # Eligible coin encountered a third time, another for the second time and one non eligible
+    initial_conditions = [
+        [ConditionOpcode.CREATE_COIN, IDENTITY_PUZZLE_HASH, 1],
+        [ConditionOpcode.CREATE_COIN, IDENTITY_PUZZLE_HASH, 2],
+    ]
+    initial_solution = SerializedProgram.from_program(Program.to(initial_conditions))
+    second_conditions = [[ConditionOpcode.CREATE_COIN, IDENTITY_PUZZLE_HASH, 3]]
+    second_solution = SerializedProgram.from_program(Program.to(second_conditions))
+    saved_cost = uint64(44)
+    created_coins = {Coin(TEST_COIN_ID, IDENTITY_PUZZLE_HASH, 1), Coin(TEST_COIN_ID, IDENTITY_PUZZLE_HASH, 2)}
+    dedup_coin_spends = {
+        TEST_COIN_ID: DedupCoinSpend(solution=initial_solution, cost=saved_cost, additions=created_coins),
+        TEST_COIN_ID2: DedupCoinSpend(solution=second_solution, cost=None, additions=set()),
+    }
+    sb1 = spend_bundle_from_conditions(TEST_COIN, initial_conditions)
+    sb2 = spend_bundle_from_conditions(TEST_COIN2, second_conditions)
+    sb3_conditions = [[ConditionOpcode.AGG_SIG_UNSAFE, G1Element(), IDENTITY_PUZZLE_HASH]]
+    sb3 = spend_bundle_from_conditions(TEST_COIN3, sb3_conditions)
+    sb = SpendBundle.aggregate([sb1, sb2, sb3])
+    mempool_item = mempool_item_from_spendbundle(sb)
+    spends_to_dedup, cost_saving, new_dedups, dedup_additions = find_duplicate_spends(mempool_item, dedup_coin_spends)
+    expected_spends_to_dedup = []
+    expected_spends_to_dedup.extend(sb1.coin_spends)
+    expected_spends_to_dedup.extend(sb2.coin_spends)
+    assert spends_to_dedup == expected_spends_to_dedup
+    saved_cost2 = uint64(44)
+    created_coins2 = {Coin(TEST_COIN_ID2, IDENTITY_PUZZLE_HASH, 3)}
+    expected_new_dedups = {TEST_COIN_ID2: DedupCoinSpend(second_solution, saved_cost2, created_coins2)}
+    assert cost_saving == saved_cost + saved_cost2
+    assert new_dedups == expected_new_dedups
+    assert dedup_additions == created_coins.union(created_coins2)
```

Comment:
I think it's clearer if you specify exactly what you expect the result to be, rather than computing it with `union`. There are only 3 items involved, right?

Reply:
Done.